qcodes/config/config.py

```diff
@@ -121,26 +123,35 @@ def update_config(self):
         """
         config = copy.deepcopy(self.defaults)
         self.current_schema = copy.deepcopy(self.defaults_schema)
-
-        if os.path.isfile(self.home_file_name):
-            home_config = self.load_config(self.home_file_name)
-            config = update(config, home_config)
-            self.validate(config, self.current_schema,
-                          self.schema_home_file_name)
-
-        if os.path.isfile(self.env_file_name):
-            env_config = self.load_config(self.env_file_name)
-            config = update(config, env_config)
-            self.validate(config, self.current_schema,
-                          self.schema_env_file_name)
-
-        if os.path.isfile(self.cwd_file_name):
-            cwd_config = self.load_config(self.cwd_file_name)
-            config = update(config, cwd_config)
+        self.config_file_path = path or self.config_file_path
+        if self.config_file_path is not None:
```

Comment:
also, what about test? (see `qcodes/tests/test_config.py`)

Reply:
Unfortunately beyond my scope. perhaps someone (the author of the existing test?) could advise me what I need to add and I can put it in this pr? Otherwise realistically I won't be able to do this in the near future.

Reply:
The author is long gone, unfortunately. I agree that `test_config.py` is not the most immediately readable piece of code. I can help you extend the test to cover the new functionality introduced in this PR.

Reply:
thanks @WilliamHPNielsen. How about I test this out when I'm next measuring and then we merge it and @WilliamHPNielsen writes the test in another pr for the actual test?